test/e2e/autoscaling_utils.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	seconds     = 10
+	sleepTime   = time.Duration(10) * time.Second
+	requestSize = 100
+)
+
+// ConsumingRC is a tool for testing. It helps create specified usage of CPU or memory (Warnig: memory not supported)
+// typical use case:
+// rc.ConsumeCPU(600)
+// // ... check your assumption here
+// rc.ConsumeCPU(300)
+// // ... check your assumption here
+type ConsumingRC struct {
+	name          string
+	milicores     int
+	mutex         *sync.Mutex
```

Comment:
Here was a comment but I apparently didn't click on "Send".

Why do we need this mutex at all? If this is only for testing can we have this single-threaded or using classic go synchronization (via channels).